### PR TITLE
add custom form for labs

### LINKED ISSUE #65: Result form(s) for labs
Right now there is a generic result field for labs.  At some point we should support generation of result forms like the following:
screen shot 2015-04-06 at 11 41 05 am

### PR DESCRIPTION
**Pull request by baoqchau**
Fixes #65 

**Changes proposed in this pull request:**
- Add custom form for lab's report.

cc @HospitalRun/core-maintainers

### CODE REVIEW

File: app/labs/edit/route.js [hunk 1]
```diff
@@ -11,13 +11,22 @@ export default AbstractEditRoute.extend(AddToPatientRoute, ChargeRoute, PatientL
...
   pricingCategory: 'Lab',
+  customForms: Ember.inject.service(),
 
   actions: {
     returnToAllItems() {
       this.controller.send('returnToAllItems');
     }
   },
 
+  getCustomFormData() {
```
Comment (jkleinsc) [inline app/labs/edit/route.js]: Looks like this code isn't used, but you need this code for forms that are set to "Always Include".  Custom forms that are set to this get added automatically on new items.

File: app/labs/edit/template.hbs [hunk 2]
```diff
@@ -42,6 +42,7 @@
     {{#if canComplete}}
       {{em-input property="result" label=(t 'labels.result') class="test-result-input"}}
     {{/if}}
+    {{custom-form-manager model=model formType="lab"}}
```
Comment (jkleinsc) [inline app/labs/edit/template.hbs]: I would move this down after notes

File: app/locales/en/translations.js [hunk 3]
```diff
@@ -175,7 +175,8 @@ export default {
         textarea: 'Large Text',
         visitFormType: 'Visit',
         opdReportFormType: 'Outpatient Report',
-        dischargeReportFormType: 'Discharge Report'
+        dischargeReportFormType: 'Discharge Report',
+        labFormType: 'Lab Form Type'
```
Comment (jkleinsc) [inline app/locales/en/translations.js]: Translated text should just be 'Lab', not 'Lab Form Type'

File: app/labs/edit/route.js (hunk 1 above)
Comment (baoqchau) [inline app/labs/edit/route.js]: So how should I fix this @jkleinsc ? You have any suggestion?

Comment (jkleinsc) [inline app/labs/edit/route.js]: Take a look at how visit does this:
https://github.com/HospitalRun/hospitalrun-frontend/blob/master/app/visits/edit/route.js#L35-L43

Basically what you need to do is to create a newData object and make sure it includes a custom forms object and then you return  customForms.setDefaultCustomForms(['lab'], newData) from the getNewData function.

**Comment by jkleinsc** [comment]
Looks good to me @baoqchau.  Thanks for the PR!

**Pull request merged**